Snapchain/cdk-validium-contracts
# Feat: SNA-287 zkvalidium integration

## Summary

* Build two snapchain images
  * `snapchain/geth-cdk-validium-contracts`
  * `snapchain/dac-setup-committee`
* Add the setup committee script
   * In the future, use another repo to reduce the docker image's (`snapchain/dac-setup-committee`) size

## Test

1. Build and push the mock-l1-network image
    * `npm i`
    * `npm run docker:contracts`
    * `npm run push:docker:contracts`
2. Build and push setup dac image
    * `npm run docker:dac:setup`
    * `npm run push:docker:dac:setup`
3. Run docker and check logs
    * `cd docker`
    * `docker-compose up -d`
    * `docker-compose logs dac-setup-committee`
        ```
		dac-setup-committee           | 
		dac-setup-committee           | > cdk-validium-contracts@0.0.1 setup:committee
		dac-setup-committee           | > node docker/scripts/setup-committee.js
		dac-setup-committee           | 
		dac-setup-committee           | DAC_URL: http://supernets2-data-availability:8444
		dac-setup-committee           | DAC_ADDRESS: 0x70997970c51812dc3a010c7d01b50e0d17dc79c8
		dac-setup-committee           | JSONRPC_HTTP_URL: http://supernets2-mock-l1-network:8545
		dac-setup-committee           | Using pvtKey deployer with address:  0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
		dac-setup-committee           | Committee seted up with 0x70997970c51812dc3a010c7d01b50e0d17dc79c8
		dac-setup-committee           | Transaction hash: 0x5d6d090ff6cd632802aa8456acc3f34804fefa8fd87d41ad09e2015b5c661695
		dac-setup-committee           | Transaction confirmed in block: 98        
    * `docker-compose down`
